src/sage/matroids/database_collections.py

```diff
@@ -195,24 +197,20 @@ def AllMatroids(n, r=None, type="all"):
             rp = min(r, n - r) if (type != "unorientable") else r
             type_file = "all" if (type != "unorientable") else "unorientable"
             file = os.path.join(
-                str(SAGE_EXTCODE), "matroids", "database",
+                str(SAGE_SRC), "sage", "ext_data", "matroids", "database",
```

Comment:
Using `SAGE_SRC` is not the way to avoid using `SAGE_EXTCODE`. The idea of #33037 is that one should not use neither `SAGE_LIB` nor `SAGE_SRC`, and `SAGE_EXTCODE`.

Instead, one is supposed to use `importlib.resources` to access package data (which I guess should be placed in the package that owns the data instead of the generic `src/sage/ext_data` although I guess you could still access files in there via `sage.ext_data`).

Reply:
That's right, using `SAGE_SRC` should also be avoided completely because this variable does not have a meaningful definition in the presence of PEP 420 namespace packages.